### PR TITLE
C2PA-54: Support for C2PA v0.1 table

### PR DESCRIPTION
**Pull request by brogdonm**
# JIRA Issue

- C2PA-54

# Checklist

- [x] **PR labeled** appropriately
- [ ] **Documentation** updated:
  - [ ] Developer documentation (ReadMe files)
  - [ ] Product documentation (Release notes, PDK Guide, Functional Spec, API documents, ...)
- [ ] **Test case(s)** added
  - [ ] Unit Tests

# Notes for Reviewers

Adds support for the C2PA v0.1 table to fonts. Should be able to read, write, remove table from a font file.

# Verification Instructions

Step into the `rust-font-tools/fonttools-cli` folder and run:

```shell
> cargo run --bin ttf-add-c2pa-table -- --help
ttf-add-c2pa-table 
Adds a C2PA v0.1 table to a font.

USAGE:
    ttf-add-c2pa-table.exe [FLAGS] [ARGS]

FLAGS:
    -h, --help       Prints help information
    -r, --remove     Remove the C2PA table from the font
    -x, --replace    Even if the font contains a C2PA table, replace with new empty table
    -V, --version    Prints version information

ARGS:
    <INPUT>     Sets the input file to use
    <OUTPUT>    Sets the output file to use

# Try out some of the scenarios, for example one may add a new C2PA table by running:
> cargo run --bin ttf-add-c2pa-table -- C:\Path\To\Input-Font.otf  C:\Path\To\Output-Font.otf

# Or one may replace a C2PA table by running:
> cargo run --bin ttf-add-c2pa-table -- --replace C:\Path\To\Input-Font.otf  C:\Path\To\Output-Font.otf

# And finally the C2PA table could be removed from the font
> cargo run --bin ttf-add-c2pa-table -- --remove C:\Path\To\Input-Font.otf  C:\Path\To\Output-Font.otf
```

The unit tests may be ran to verify as well:

```shell
> cd rust-font-tools/fonttools-rs
rust-font-tools/fonttools-rs> cargo test
```

> Actively maintained by the @Monotype/driverpdldev team.

### CODE REVIEW

File: fonttools-rs/src/tables/C2PA.rs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+#[allow(non_snake_case)]
+pub struct C2PA {
+    /// Major version of the C2PA table record
+    pub majorVersion: uint16,
+    /// Minor version of the C2PA table record
+    pub minorVersion: uint16,
+    /// Optional URI to an active manifest
+    pub activeManifestUri: Option<String>,
+    /// Optional embedded manifest store
+    pub c2paManifestStore: Option<String>,
```
Comment (MikeJonesMT) [inline fonttools-rs/src/tables/C2PA.rs]: This should probably use a reference to binary data stored as [u8] ?

And are you going to drop the `c2pa` prefix on this?  Maybe call it `manifestStore` ?

**Review by MikeJonesMT — CHANGES_REQUESTED** [review]
Do you want to drop `c2pa` prefix for names that are scoped inside of a `c2pa` thingus?

File: fonttools-rs/src/tables/C2PA.rs (hunk 1 above)
Comment (brogdonm) [reply fonttools-rs/src/tables/C2PA.rs]: Changed to a `Option<Box<[u8]>>` and added an accessor `get_manifest_store(&self) -> Option<&[u8]>` to address.

**Comment by brogdonm** [comment]
> Do you want to drop `c2pa` prefix for names that are scoped inside of a `c2pa` thingus?

Agreed, updated the specification document and updated the code as well. 

Additionally, changed the manifest store representation to a byte array option.

**Review by MikeJonesMT — APPROVED** [review]
I like it.  The executable to add tables and remove is super nice.  Looks fantastic to this noob.

**Pull request merged**